metagraph/CMakeLists.txt

```diff
@@ -319,6 +319,8 @@ IF(APPLE)
     set(THREADS_PREFER_PTHREAD_FLAG ON)
 ENDIF()
 add_subdirectory(external-libraries/spdlog)
+target_compile_options(spdlog PRIVATE -Wno-dangling-reference)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-dangling-reference>)
```

Comment:
I'm not sure what's the best course of action here is.

Basically, on newer g++ version, spdlog throws the following warning:
```console
/home/adamant/metagraph/metagraph/external-libraries/spdlog/include/spdlog/fmt/bundled/core.h:2886:44: in ‘constexpr’ expansion of ‘fmt::v8::make_format_args<>(((const std::__cxx11::basic_string<char>&)args#0), ((const long unsigned int&)args#1), ((const std::__cxx11::basic_string<char>&)args#2))’
/home/adamant/metagraph/metagraph/external-libraries/spdlog/include/spdlog/fmt/bundled/core.h:1566:15: error: possibly dangling reference to a temporary [-Werror=dangling-reference]
1566 | const auto& arg = arg_mapper<Context>().map(val);
| ^~~
/home/adamant/metagraph/metagraph/external-libraries/spdlog/include/spdlog/fmt/bundled/core.h:1566:46: note: the temporary was destroyed at the end of the full expression ‘fmt::v8::detail::arg_mapper<fmt::v8::basic_format_context<fmt::v8::appender, char> >().fmt::v8::detail::arg_mapper<fmt::v8::basic_format_context<fmt::v8::appender, char> >::map<std::__cxx11::basic_string<char> >((* & val))’
1566 | const auto& arg = arg_mapper<Context>().map(val);
| ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
cc1plus: all warnings being treated as errors
```

Unfortunately it's due to code in a header file, meaning that it also affects the compilation of metagraph source files. I've also tried update spdlog version, but it produces some other kinds of errors starting at `v1.14.0`.